processor/trackingplan.go

```diff
@@ -102,7 +102,7 @@ func (proc *Handle) validateEvents(groupedEventsByWriteKey map[WriteKeyT][]trans
 		trackingPlanEnabledMap[SourceIDT(sourceID)] = true
 
 		var successMetrics []*types.PUReportedMetric
-		eventsToTransform, successMetrics, _, _ := proc.getDestTransformerEvents(response, commonMetaData, destination, transformer.TrackingPlanValidationStage, true, false) // Note: Sending false for usertransformation enabled is safe because this stage is before user transformation.
+		eventsToTransform, successMetrics, _, _ := proc.getDestTransformerEvents(response, commonMetaData, eventsByMessageID, destination, transformer.TrackingPlanValidationStage, true, false) // Note: Sending false for usertransformation enabled is safe because this stage is before user transformation.
```

Comment:
Consider using another function for tracking plan without disturbing the flow for userTransform and destTransformation

Reply:
I think we should start showing sample event for success state in userTransform and destTransformation too. Is there any reason behind not showing it. 
From reporting side, this might increase computation while generating hourly view data. they have asked to hold sending this data until timescale upgrade task is done. For this reason I have used the same existing function.